### PR TITLE
Blog App - Processing Data in models 🤖 

### PR DESCRIPTION
**Pull request by evansnyamekye**
### New Features Implementation 

**Create models**

- Create model classes for all entities as shown in the [ERD diagram].
- Set up associations between models.
   Remember that the `author_id` column in the `posts` table should be the foreign_key for the `users` table.

**Use models to insert data**

- Familiarize yourself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).
- Open the rails console for your project.
- Create at least one user by running the following code:
   `first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
   second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')`
- Create at least 4 posts written by one of the users you created by running the following code:
   `first_post = Post. create(author: first_user, title: 'Hello', text: 'This is my first post')`
- Create at least 6 post comments for one of the posts you created by running the following code:
   `Comment.create(post: first_post, user: second_user, text: 'Hi Tom!' )`
- Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog- 
   app/lessons/active_record_crud.md) to find, update, and delete entities.

**Create custom methods**
Your models should include some additional methods.
**Users**
A method that returns the 3 most recent posts for a given user.
**Posts**
A method that updates the posts counter for a user.
A method that returns the 5 most recent comments for a given post.
**Comments**
A method that updates the comments counter for a post.
**Likes**
A method that updates the likes counter for a post.
Go to Rails C and check if your methods are working.

Below are the screenshot data I worked with. 
1.
ps1
2.
ps2
3.
ps3
4.
ps4
5.
ps5
6.
ps6
7.
ps7
8.
ps8
9.
ps9
10.
ps10

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,3 @@
 class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id'
```
Comment (thecodechaser) [inline app/models/user.rb:2]: - [ ] A user also has an association with comments and likes, Please make sure to fix this issue so associations between models work as expected.
  - You can consider using this code.
```ruby
 has_many :comments
 has_many :likes

File: app/models/post.rb [hunk 2]
```diff
@@ -1,2 +1,3 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
```
Comment (thecodechaser) [inline app/models/post.rb:2]: - [ ] A post also has an association with comments and likes, Please make sure to fix this issue so associations between models work as expected.
  - You can consider using this code.
```ruby
 has_many :comments
 has_many :likes

File: app/models/user.rb (hunk 1 above)
Comment (thecodechaser) [inline app/models/user.rb:2]: - [ ] You didn't implement the custom methods for any of the models. Please make sure to implement all custom methods as requested in the project requirements.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @evansnyamekye👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job with model associations. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (evansnyamekye) [reply app/models/user.rb:2]: Hi, @thecodechaser Fixed all required changes as requested. 

Thank you 😺.

**Review by IngDavidTM — APPROVED** [review]
Hi @evansnyamekye 🙋‍♂️, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**